alpine-nix-rails-nginx/Dockerfile

```diff
@@ -0,0 +1,53 @@
+# dkubb/alpine-nix-rails-nginx
+
+FROM dkubb/alpine-nix:latest
+MAINTAINER Dan Kubb <dkubb@fastmail.com>
+
+EXPOSE 80
+
+# Configure nix
+COPY config.nix /root/.nixpkgs/config.nix
+
+# Install system dependencies
+RUN nix-env --install --fallback \
+    acl-2.2.52 \
+    bundler-1.10.5 \
+    gcc-wrapper-5.2.0 \
+    git-2.5.0 \
+    gnumake-4.1 \
+    libressl-2.2.2 \
+    libxml2-2.9.2 \
+    libxslt-1.1.28 \
+    nginx-1.8.0 \
+    nodejs-0.12.7 \
+    ruby-2.2.2-p0 \
+    runit-2.1.2 \
+    zlib-1.2.8 \
+  && nix-collect-garbage --delete-old \
+  && nix-store --optimize
+
+RUN adduser -DS nginx
+RUN adduser -DS rails
+
+# Setup system configuration
+COPY etc /etc/
+
+# Create system directories and service symlinks
+RUN mkdir -m 1750 -- /opt \
+  && chown root:nogroup /opt \
```

Comment:
The `--` terminator has to be explicitly handled within the script. It's not something you get for free from the shell. I wanted to keep the argument handling in `setup-directories.sh` as simple as possible because I find `getopt` and `getopts` to be overkill for simple scripts.

With that said, when a utility does support `--` I try to use it to make a clear separation between options and positional arguments.